source/git/source.go

```diff
@@ -207,6 +209,9 @@ func (gs *gitSourceHandler) shaToCacheKey(sha, ref string) string {
 	if gs.src.Subdir != "" {
 		key += ":" + gs.src.Subdir
 	}
+	if gs.src.CommitHash != "" {
```

Comment:
Why is this needed? Isn't `sha` already same as `CommitHash`?

Reply:
No, `CommitHash` can be a partial hash

Reply:
Why does that matter, if we already have a complete one.

Same as https://github.com/moby/buildkit/pull/5903/files/076850aeb7720b56683710d8e30b05fbb9352b97#diff-bf1789b0b19c8d6533a10da5e27f0b31da2cdcd37510a787498c3d19a985d470 . Eg. if you build from `alpine:latest` or `alpine:latest@sha256` then if the digest matches then they will generate the same cache key.